src/components/settings/settings-custom-labels/settings-custom-labels.js

```diff
@@ -0,0 +1,203 @@
+          displayLabel: PropTypes.string,
+          displayOnFullTextFinder: PropTypes.bool,
+          displayOnPublicationFinder: PropTypes.bool,
+          id: PropTypes.number,
+        })),
+      }),
+    }).isRequired,
+    updateCustomLabels: PropTypes.func.isRequired,
+  };
+
+  state = {
+    modalIsOpen: false,
+    removingLabels: '',
+  }
+
+  prepareInitialValues = (data) => {
+    const allCustomLabels = getAllCustomLabelsFields(data);
+    const mappedCustomLabels = allCustomLabels.map(({ attributes }) => {
+      const {
+        id,
+        displayLabel,
+        displayOnFullTextFinder,
+        displayOnPublicationFinder,
+      } = attributes;
+
+      return {
+        [`${id}displayLabel`]: displayLabel,
+        [`${id}displayOnFullTextFinder`]: displayOnFullTextFinder,
+        [`${id}displayOnPublicationFinder`]: displayOnPublicationFinder,
+      };
+    });
+
+    return mappedCustomLabels.reduce((acc, customLabel) => ({ ...acc, ...customLabel }), {});
+  };
+
+  removeCustomLabels = ({ values }) => {
+    const { updateCustomLabels } = this.props;
+    const formattedFormValues = formatCustomLabelsValues(values);
+
+    updateCustomLabels(formattedFormValues);
+    this.closeModal();
+  }
+
+  getRemovingCustomLabels = (formValues, initialValues) => {
+    const formDisplayLabels = getDisplayLabels(formValues);
+    const initialDisplayLabels = getDisplayLabels(initialValues);
+    const removingDisplayLabels = pickBy(initialDisplayLabels, (value, key) => (formDisplayLabels[key] === undefined));
+
+    return Object.values(removingDisplayLabels).join(', ');
+  }
+
+  onSubmit = (formValues, { getState }) => {
+    const { updateCustomLabels } = this.props;
+    const { initialValues } = getState();
+
+    const removingLabels = this.getRemovingCustomLabels(formValues, initialValues);
+
+    if (removingLabels.length !== 0) {
+      this.setState({
+        modalIsOpen: true,
+        removingLabels,
+      });
+    } else {
+      const formattedFormValues = formatCustomLabelsValues(formValues);
+
+      updateCustomLabels(formattedFormValues);
+    }
+  }
+
+  getToastLabels() {
+    const {
+      customLabels: {
+        errors,
+        isUpdated,
+      },
+    } = this.props;
+
+    if (isUpdated) {
+      return [{
+        message: <FormattedMessage id='ui-eholdings.settings.customLabels.toast' />,
+        type: 'success',
+        id: `update-custom-labels-success-${Date.now()}`
+      }];
+    } else {
+      return errors.map((error) => ({
+        message: error.title,
+        type: 'error',
+      }));
+    }
+  }
+
+  closeModal = () => this.setState({ modalIsOpen: false });
+
+  render() {
+    const {
+      confirmUpdate,
+      customLabels,
+    } = this.props;
+    const {
+      modalIsOpen,
+      removingLabels,
+    } = this.state;
+
+    const customLabelsData = get(customLabels, ['items', 'data'], []);
+    const allCustomLabels = getAllCustomLabelsFields(customLabelsData);
```

Comment:
Please, remove this function call since `getAllCustomLabelsFields` is already used in `prepareInitialValues`